### PR TITLE
implement useReassemble

### LINKED ISSUE #60: useReassemble

### PR DESCRIPTION
**Pull request by SahandAkbarzadeh**
resolves #60

### CODE REVIEW

**Comment by rrousselGit** [comment]
Hello, sorry for the late reply!

Thanks for your contribution. Before merging this, we'll want to add tests first.

**Comment by SahandAkbarzadeh** [comment]
@rrousselGit i have to learn more about flutter tests, it may take a little bit of time if it's ok ?

**Comment by rrousselGit** [comment]
No problem, take your time

**Comment by SahandAkbarzadeh** [comment]
@rrousselGit fixed the problem that tests didn't register, it should be ok now.

File: test/use_reassemble_test.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }));
+
+    verifyNoMoreInteractions(reassemble);
+
+    hotReload(tester);
+    await tester.pump();
+
+    verifyInOrder([
+      reassemble.call(),
+    ]);
```
Comment (rrousselGit) [inline test/use_reassemble_test.dart]:
```suggestion
    verify(reassemble()).called(1);
```

File: lib/src/misc.dart [hunk 2]
```diff
@@ -110,3 +110,31 @@ class _PreviousHookState<T> extends HookState<T, _PreviousHook<T>> {
...
   T build(BuildContext context) => previous;
 }
+
+/// Runs the callback on every hot reload
+/// similar to reassemble in the Stateful widgets
+/// See also:
+///
+///  * [State.reassemble]
+void useReassemble(VoidCallback callback) =>
+    Hook.use(_ReassembleHook(callback));
```
Comment (rrousselGit) [inline lib/src/misc.dart]: It's probably a good idea to wrap that `Hook.use` into an `assert`. This way, the hook disappears in release mode

**Comment by rrousselGit** [comment]
A few extra small changes and it should be good to go :)

Thanks for your contribution!

File: lib/src/misc.dart [hunk 3]
```diff
@@ -110,3 +110,31 @@ class _PreviousHookState<T> extends HookState<T, _PreviousHook<T>> {
   @override
   T build(BuildContext context) => previous;
 }
+
+/// Runs the callback on every hot reload
+/// similar to reassemble in the Stateful widgets
+/// See also:
```
Comment (rrousselGit) [inline lib/src/misc.dart]:
```suggestion
///
/// See also:
```

File: lib/src/misc.dart (hunk 2 above)
Comment (SahandAkbarzadeh) [reply lib/src/misc.dart]: by doing that i have to change `Hook<void>` to `Hook<bool>` ( or other types ) is it ok or is there a better way?

**Comment by rrousselGit** [comment]
Perfect. Thanks!

**Pull request merged**